contentctl/api.py

```diff
@@ -0,0 +1,124 @@
+from pathlib import Path
+from typing import Any, Union, Type
+#from contentctl import contentctl
+from contentctl.input.yml_reader import YmlReader
+from contentctl.objects.config import test_common, test_servers, test
+
+
+
+def configFromFile(path:Path=Path("contentctl.yml"), config: dict[str,Any]={}, 
+                   configType:Type[Union[test,test_servers]]=test)->Union[test,test_servers]:
+    
+    """
+    Fetch a configuration object that can be used for a number of different contentctl
+    operations including validate, build, inspect, test, and test_servers. A file will 
+    be used as the basis for constructing the configuration.
+
+    Args:
+        path (Path, optional): Relative or absolute path to a contentctl config file. 
+        Defaults to Path("contentctl.yml"), which is the default name and location (in the current directory)
+        of the configuration files which are automatically generated for contentctl.
+        config (dict[], optional): Dictionary of values to override values read from the YML
+        path passed as the first argument. Defaults to {}, an empty dict meaning that nothing
+        will be overwritten 
+        configType (Type[Union[test,test_servers]], optional): The Config Class to instantiate. 
+        This may be a test or test_servers object. Note that this is NOT an instance of the class. Defaults to test.
+    Returns:
+        test_common: Returns a complete contentctl test_common configuration. Note that this configuration
+        will have all applicable field for validate and build as well, but can also be used for easily
+        construction a test or test_servers object.  
+    """    
+
+    try:
+        yml_dict = YmlReader.load_file(path, add_fields=False)
+        
+        
+    except Exception as e:
+        raise Exception(f"Failed to load contentctl configuration from file '{path}': {str(e)}")
+    
+    #Apply settings that have been overridden from the ones in the file
+    try:
+        yml_dict.update(config)
+    except Exception as e:
+        raise Exception(f"Failed updating dictionary of values read from file '{path}'"
+                        f" with the dictionary of arguments passed: {str(e)}")
+
+    # The function below will throw its own descriptive exception if it fails
+    configObject = configFromDict(yml_dict, configType=configType)
+
+    return configObject
+
+
+
+
+def configFromDict(config: dict[str,Any]={}, 
+                   configType:Type[Union[test,test_servers]]=test)->Union[test,test_servers]:
```

Comment:
Could the return type not just be `test_common` since `test` and `test_servers` both inherit `test_common`?

Reply:
Yes - both functions that return configs now return a test_common.
We will leave it up to type-checking in the IDE + users to ensure that if they are accessing fields unique to a test or test_servers object that they have requested (and returned) the correct object type.